pkg/meta/boltdb/boltdb.go

```diff
@@ -1034,464 +929,476 @@ func (bdw *BoltDB) DeleteSignature(repo string, signedManifestDigest godigest.Di
-				continue
-			}
+		err := proto.Unmarshal(repoMetaBlob, &repoMeta)
+		if err != nil {
+			return err
+		}
 
-			var repoMeta mTypes.RepoMetadata
+		repoMeta.Stars++
 
-			err := json.Unmarshal(repoMetaBlob, &repoMeta)
-			if err != nil {
-				return err
-			}
+		repoMetaBlob, err = proto.Marshal(&repoMeta)
+		if err != nil {
+			return err
+		}
 
-			rank := common.RankRepoName(searchText, repoMeta.Name)
-			if rank == -1 {
-				continue
-			}
+		return buck.Put([]byte(repo), repoMetaBlob)
+	})
 
-			repoMeta.IsBookmarked = zcommon.Contains(userBookmarks, repoMeta.Name)
-			repoMeta.IsStarred = zcommon.Contains(userStars, repoMeta.Name)
-			repoMeta.Rank = rank
+	return err
+}
 
-			for tag, descriptor := range repoMeta.Tags {
-				switch descriptor.MediaType {
-				case ispec.MediaTypeImageManifest:
-					manifestDigest := descriptor.Digest
+func (bdw *BoltDB) DecrementRepoStars(repo string) error {
+	err := bdw.DB.Update(func(tx *bbolt.Tx) error {
+		buck := tx.Bucket([]byte(RepoMetaBuck))
 
-					manifestMeta, err := fetchManifestMetaWithCheck(repoMeta, manifestDigest,
-						manifestMetadataMap, manifestBuck)
-					if err != nil {
-						return fmt.Errorf("metadb: error fetching manifest meta for manifest with digest %s %w",
-							manifestDigest, err)
-					}
+		repoMetaBlob := buck.Get([]byte(repo))
+		if repoMetaBlob == nil {
+			return zerr.ErrRepoMetaNotFound
+		}
 
-					manifestMetadataMap[descriptor.Digest] = manifestMeta
-				case ispec.MediaTypeImageIndex:
-					indexDigest := descriptor.Digest
+		var repoMeta proto_go.RepoMeta
 
-					indexData, err := fetchIndexDataWithCheck(indexDigest, indexDataMap, indexBuck)
-					if err != nil {
-						return fmt.Errorf("metadb: error fetching index data for index with digest %s %w",
-							indexDigest, err)
-					}
+		err := proto.Unmarshal(repoMetaBlob, &repoMeta)
+		if err != nil {
+			return err
+		}
 
-					var indexContent ispec.Index
+		if repoMeta.Stars == 0 {
+			return nil
+		}
 
-					err = json.Unmarshal(indexData.IndexBlob, &indexContent)
-					if err != nil {
-						return fmt.Errorf("metadb: error while unmashaling index content for %s:%s %w",
-							repoName, tag, err)
-					}
+		repoMeta.Stars--
 
-					for _, manifest := range indexContent.Manifests {
-						manifestDigest := manifest.Digest
+		repoMetaBlob, err = proto.Marshal(&repoMeta)
+		if err != nil {
+			return err
+		}
 
-						manifestMeta, err := fetchManifestMetaWithCheck(repoMeta, manifestDigest.String(),
-							manifestMetadataMap, manifestBuck)
-						if err != nil {
-							return err
-						}
+		return buck.Put([]byte(repo), repoMetaBlob)
+	})
 
-						manifestMetadataMap[manifest.Digest.String()] = manifestMeta
-					}
+	return err
+}
 
-					indexDataMap[indexDigest] = indexData
-				default:
-					bdw.Log.Error().Str("mediaType", descriptor.MediaType).Msg("Unsupported media type")
+func (bdw *BoltDB) GetRepoStars(repo string) (int, error) {
```

Comment:
Is this function properly used and tested? It seems to be not covered. 

If I remember correctly the calls were removed from the test in the other file.